pkg/cmd/promotelearner.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/controllers"
+	"github.com/openshift/microshift/pkg/version"
+	"github.com/spf13/cobra"
+
+	"k8s.io/klog/v2"
+)
+
+func NewPromoteLearnerCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "promote-learner",
+		Short: "Promote learner node to member in the etcd cluster",
+		Long: `This command promotes the local etcd instance from a learner node to a full voting member within an existing MicroShift etcd cluster.
+It:
+  - Connects to the etcd cluster using the current node's configuration.
+  - Verifies that the local etcd instance is currently a learner.
+  - Issues a promote request to the cluster.
+  - Restarts the MicroShift service to make the membership change effective to apiserver.
+Use this command only after the learner node has fully caught up with the cluster and you are ready for it to become a voting member.
```

Comment:
Maybe we should add check in code to do that? Before the step 3: "Issues a promote request to the cluster"